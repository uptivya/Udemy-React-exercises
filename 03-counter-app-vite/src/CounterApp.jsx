import {useState} from 'react'
import PropTypes from "prop-types";


export const CounterApp = ( {value} ) => {
    
    const [counter, setCounter] = useState(value);
    const handleAdd = () => setCounter(counter + 1);
    const handleMinus = () => setCounter(counter - 1);
    const handleReset = () => setCounter(value);
    
  return (
    <>
        <h1>CounterApp</h1>
        <p data-testid="initial-counter">{ counter }</p>
        <button aria-label="sub1" onClick={ handleMinus }>-1</button>
        <button aria-label="reset" onClick={ handleReset }>Reset</button>
        <button aria-label="add1" onClick={ handleAdd }>+1</button>
    </>
  )
}

CounterApp.propTypes = {
     value: PropTypes.number
}