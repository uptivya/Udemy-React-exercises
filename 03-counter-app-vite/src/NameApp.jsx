import PropTypes from 'prop-types'

const returnName = (flag, nomRebut, cognomRebut) => {
    const frase = (flag) ? `Sí, sóc en ${nomRebut+' '+cognomRebut}` : `No, no sóc en ${nomRebut+' '+cognomRebut}`
    return frase
}

const returnSubtitle = (flag) => (flag) ? 'Guapíssim' : 'Lletgíssim';

export const Name = ({nom, cognom, flag}) => {
    return (
        <>
            <h1 data-testid="h1test">{returnName(flag,nom,cognom)}</h1>
            <p>{returnSubtitle(flag)}</p>
        </>
    )
}

Name.propTypes = {
    nom: PropTypes.string.isRequired,
    flag: PropTypes.bool.isRequired,
    cognom: PropTypes.string
}

Name.defaultProps = {
    nom: 'No hi ha cap nom.',
    flag: false,
    cognom: '',
}
