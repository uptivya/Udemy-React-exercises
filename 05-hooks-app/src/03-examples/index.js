export * from './Loading'
export * from './QuoteItem'