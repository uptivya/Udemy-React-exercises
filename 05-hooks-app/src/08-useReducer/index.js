export * from './AddTodoForm'
export * from './TodoItem'
export * from './TodoList'
export * from './todoReducer'
