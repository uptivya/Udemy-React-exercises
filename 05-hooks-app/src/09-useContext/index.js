export * from './ContactPage'
export * from './HomePage'
export * from './LoginPage'
export * from './404'
export * from './Navbar'