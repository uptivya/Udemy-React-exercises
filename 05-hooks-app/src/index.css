body{
    padding: 20px;
    margin: 0;
    background-color: blanchedalmond;
}

button{
    margin-right: 10px;
}

p.blockquote{
    display: inline-block;
}