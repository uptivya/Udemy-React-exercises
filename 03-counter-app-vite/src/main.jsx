import React from 'react';
import ReactDOM from "react-dom/client";
import { CounterApp } from './CounterApp';
/* import { Name } from "./NameApp"; */

import './styles.css';

ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        <CounterApp value={20} />
        {/* <Name nom='Dani' cognom="Morales" flag={true} /> */}

    </React.StrictMode>
)