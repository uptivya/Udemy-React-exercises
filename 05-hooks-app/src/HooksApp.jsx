import { CounterApp } from "./01-useState/CounterApp"

export const HooksApp = () => {
  return (
    <>
        <h1>HooksApp</h1>
        
        {/* Counter App */}
        <CounterApp />
    </>
  )
}
