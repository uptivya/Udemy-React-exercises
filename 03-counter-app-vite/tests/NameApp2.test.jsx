import { render,screen } from "@testing-library/react";
import { Name } from "../src/NameApp";

describe('Pruebas en <NameApp />', () => {
    const nom = 'Dani'

    test('Debe hacer match con el snapshot', () => {
        const { container } = render( < Name nom={nom} cognom='Morales' flag={true} />)
        expect(container).toMatchSnapshot();
        //console.log(container);
    })
    
    test('Debe contener el tÃ­tulo', () => {
        render( < Name nom={nom} cognom='Morales' flag={true} />)
        expect(screen.getByRole('heading', {level:1}).innerHTML).toContain('Dani');
    })
});