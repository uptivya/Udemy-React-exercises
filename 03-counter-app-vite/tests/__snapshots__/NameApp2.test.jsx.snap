// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <NameApp /> Debe hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="h1test"
  >
    Sí, sóc en Dani Morales
  </h1>
  <p>
    Guapíssim
  </p>
</div>
`;
